src/check_python_versions/sources/poetry_pyproject.py

```diff
@@ -0,0 +1,197 @@
+"""
+Support for pyproject.toml.
```

Comment:
But AFAIU Poetry is not the only build tool that supports pyproject.toml?  There are also things like flit, and I think even setuptools?  (I'm not sure about setuptools -- I know there's a way to use declarative configuration for setuptools with an empty/minimal setup.py, but I don't know if it goes into setup.cfg, or if it can also go into pyproject.toml.)

Reply:
I think it can be made to work fine if you make the `extract()` callback of the new Source check the value of `[build-system] build-backend` and return `None` if it doesn't match `poetry.whatever`.  Then multiple sources can declare that they support pyproject.toml and they should all cooperate nicely.

Reply:
> But AFAIU Poetry is not the only build tool that supports pyproject.toml? There are also things like flit, and I think even setuptools? (I'm not sure about setuptools -- I know there's a way to use declarative configuration for setuptools with an empty/minimal setup.py, but I don't know if it goes into setup.cfg, or if it can also go into pyproject.toml.)

yes, recently many tools are starting support pyproject.toml instead of setup.py, but the content could differ (see [setuptools](https://setuptools.pypa.io/en/latest/userguide/quickstart.html) as example, it does not start with `[tool.poetry]` section but with `[project]` )

Reply:
> I think it can be made to work fine if you make the `extract()` callback of the new Source check the value of `[build-system] build-backend` and return `None` if it doesn't match `poetry.whatever`. Then multiple sources can declare that they support pyproject.toml and they should all cooperate nicely.

surely it can be improved with other tools that are using pyproject.toml, I've to "study" your suggestion 😄